<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EditModelDialog</class>
 <widget class="QDialog" name="EditModelDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>359</width>
    <height>314</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Edit Model</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="IdLabel">
       <property name="text">
        <string>ID:</string>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="modelIdInput"/>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="NameLabel">
       <property name="text">
        <string>Name:</string>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QLineEdit" name="modelNameInput"/>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QLabel" name="compartmentLabel">
     <property name="text">
      <string>Compartments:</string>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="ElementTableView" name="compartmentTableView">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="editTriggers">
        <set>QAbstractItemView::DoubleClicked</set>
       </property>
       <attribute name="horizontalHeaderStretchLastSection">
        <bool>true</bool>
       </attribute>
       <attribute name="verticalHeaderVisible">
        <bool>false</bool>
       </attribute>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QPushButton" name="addCompartmentButton">
         <property name="text">
          <string>Add</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="deleteCompartmentButton">
         <property name="text">
          <string>Delete</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Save</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>ElementTableView</class>
   <extends>QTableView</extends>
   <header>GEMEditor.widgets.views</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>EditModelDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>304</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>EditModelDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>304</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>modelIdInput</sender>
   <signal>textChanged(QString)</signal>
   <receiver>EditModelDialog</receiver>
   <slot>activateButton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>202</x>
     <y>25</y>
    </hint>
    <hint type="destinationlabel">
     <x>371</x>
     <y>65</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>modelNameInput</sender>
   <signal>textChanged(QString)</signal>
   <receiver>EditModelDialog</receiver>
   <slot>activateButton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>178</x>
     <y>45</y>
    </hint>
    <hint type="destinationlabel">
     <x>385</x>
     <y>75</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>addCompartmentButton</sender>
   <signal>clicked()</signal>
   <receiver>EditModelDialog</receiver>
   <slot>add_compartment()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>330</x>
     <y>95</y>
    </hint>
    <hint type="destinationlabel">
     <x>380</x>
     <y>178</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>deleteCompartmentButton</sender>
   <signal>clicked()</signal>
   <receiver>EditModelDialog</receiver>
   <slot>delete_compartment()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>312</x>
     <y>128</y>
    </hint>
    <hint type="destinationlabel">
     <x>378</x>
     <y>228</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>EditModelDialog</sender>
   <signal>accepted()</signal>
   <receiver>EditModelDialog</receiver>
   <slot>save_changes()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>179</x>
     <y>156</y>
    </hint>
    <hint type="destinationlabel">
     <x>179</x>
     <y>156</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>activateButton()</slot>
  <slot>add_compartment()</slot>
  <slot>delete_compartment()</slot>
  <slot>save_changes()</slot>
 </slots>
</ui>
